avocado/core/test.py

```diff
@@ -1093,13 +1094,32 @@ def fetch_asset(self, name, asset_hash=None, algorithm=None,
         :param locations: list of URLs from where the asset can be
                           fetched (optional)
         :param expire: time for the asset to expire
-        :raise EnvironmentError: When it fails to fetch the asset
-        :returns: asset file local path
+        :param find_only: whether it should only look for the asset in the
+                          cache and do not download/move it to the cache.
+        :param cancel_on_missing: whether the test should be canceled if the
+                                  asset was not found in the cache. Takes
+                                  effect only when parameter `find_only` is
```

Comment:
I find that useful in any case... any reason to restrict it to only when `find_only` is enabled?

Reply:
In the beginning, I thought it would be better to change as little as possible the behavior of the method, but now that you commented, it makes sense, users may want to cancel the test when `fetch` throws an exception instead of handling it on the test side. I'm going to make the change and send another version.

Reply:
cool!